jaxrs/src/main/java/org/jboss/as/jaxrs/JaxrsSubsystemAdd.java

```diff
@@ -68,6 +74,23 @@ public void execute(DeploymentProcessorTarget processorTarget) {
 
         JaxrsServerConfig serverConfig = createServerConfig(operation, context);
         JaxrsServerConfigService.install(serviceTarget, serverConfig);
+
+        // If CDI is available and this is community level support, add the Jakarta REST type annotations as bean defining annotations.
+        // We currently only support community level as RESTEasy 6.2 specially handles CDI non-proxyable resources. In
+        // future versions of RESTEasy, this will not be done as the Jakarta REST specification itself will assign these
+        // annotations as @Stereotype annotations making bean defining annotations.
+        if (context.getStability() == Stability.COMMUNITY && context.hasOptionalCapability(Capabilities.WELD_CAPABILITY_NAME, null, null)) {
```

Comment:
Should this not also be available for servers using lower stability levels?
If so, I think you want:
```java
if (context.getStability().enables(Stability.COMMUNITY) && ...)
```